### PR TITLE
Adding cascading waiters to avoid the 60 minute deployment manager timeout

### PR DESCRIPTION
**Pull request by sasdaberg**

### CODE REVIEW

File: templates/vms.py [hunk 1]
```diff
@@ -109,25 +106,73 @@
...
+fi
+gcloud beta runtime-config configs variables set success1/deploy-status success --config-name $DEPLOYMENT-runtime-config &>> $SAS_VIYA_HTTP_LOG
+end=$((SECONDS+3000)) # Wait for 50 minutes
+while [[ $SECONDS -lt $end ]] && [[ $(curl -sk -o /dev/null -w "%%{http_code}" https://$LOADBALANCERIP/SASLogon/login) != 200 ]];
+do
+  echo "STATUSCODE = " $(curl -sk -o /dev/null -w "%%{http_code}" https://$LOADBALANCERIP/SASLogon/login) >> $SAS_VIYA_HTTP_LOG
+  echo "Viya is not ready yet" >> $SAS_VIYA_HTTP_LOG
+  sleep 60
+done
+if [[ $(curl -sk -o /dev/null -w "%%{http_code}" https://$LOADBALANCERIP/SASLogon/login) == 200 ]]; then
```
Comment (chlync) [inline templates/vms.py]: Looks functional, but could I get you to store the uri you are querying in a variable or preferably to move these waiter loops into a called function.

File: templates/runtimeconfig.py [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                'failure': {
+                    'cardinality': {
+                        'number': 1,
+                        'path': '/failure2'
+                    }
+                }
+            }
+        },
+        {
+            'name': "%s-startup-waiter3" % deployment,
```
Comment (chlync) [inline templates/runtimeconfig.py]: While 3 waiters is likely to catch most of the runs, for large files or times when the network is uncertain, there might be a longer delay during install. Could I get you to add an additional waiter for buffer (I have seen installs take 4 hours, rarely, but still happened).

File: templates/vms.py [hunk 3]
```diff
@@ -4,21 +4,25 @@
 
 """ Startup script for Ansible Controller """
 ansible_startup_script = '''#!/bin/bash
-# The VM needs a few seconds to come up before executing the startup scripts
-sleep 10
 ###################################
 # Setting up environment
 ###################################
 export COMMON_CODE_COMMIT="3640aae263bb808003b9e6e7d89739ea01a22635"
+export DEPLOYMENT="%s"
```
Comment (chlync) [inline templates/vms.py]: I may be a little late to the game here, but any chance I could get you to switch from % formatting to str .format(). This is a little more verbose, but you can do easier inline formatting and you can map the the strings to named kwargs, making it easier to tell what is being done.

https://pyformat.info/  <- info on python formatting with tutorials, should work in 2.7

File: templates/runtimeconfig.py (hunk 2 above)
Comment (sasdaberg) [reply templates/runtimeconfig.py]: Added Waiter 4.

File: templates/vms.py (hunk 3 above)
Comment (sasdaberg) [reply templates/vms.py]: Updated string replacement format.

File: templates/vms.py (hunk 1 above)
Comment (sasdaberg) [reply templates/vms.py]: I made a few changes to the waiter loop.  It's not a function call, but hopefully it's an improvement.

File: templates/runtimeconfig.py [hunk 4]
```diff
@@ -93,6 +93,32 @@ def GenerateConfig(context):
                     }
                 }
             }
+        },
+        {
+            'name': "%s-startup-waiter4" % deployment,
```
Comment (chlync) [inline templates/runtimeconfig.py]: looks like this waiter code did not get the changes to .format()

**Pull request merged**